[bitnami/harbor] Add parameters to configure tracing

kristinnardal2 opened the pull request:
### Description of the change

Tracing support was added to Harbor in version 2.4.0.

Currently the way to enable this with the Helm chart is to set
extraEnvVars for the core, jobservice and registry controller.

This PR adds parameters into the values.yaml file to be able to easily
enable tracing

Signed-off-by: Kristinn Björgvin Árdal <kristinnardalsecondary@gmail.com>

### Benefits

This gives us an easier way of enabling tracing, without having to
manually figure out which environment variables need to be added.

### Additional information

I've tried to keep code duplication to a minimum by templating the
environment variables and including them in the envvar configmaps for
each of the deployments. Since the registry controller did not have
a envvars configmap I added one.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

javsalgar commented:
Hi! Thanks for the PR! Could you fix the merge issue?

kristinnardal2 commented:
@javsalgar I've merged the master branch into this one and fixed the merge issue

marcosbc commented on bitnami/harbor/Chart.yaml:
```diff
@@ -34,4 +34,4 @@ sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/harbor-registryctl
   - https://goharbor.io/
-version: 15.0.4
+version: 15.0.5
```
Since this is adding a new feature, could you increase the minor version instead? I.e. to `15.1.0`.

kristinnardal2 replied:
Done

marcosbc commented on bitnami/harbor/templates/registry/registry-ctl-cm-envvars.yaml:
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
```
Shouldn't we also disable this configmap if there are no configurations enabled? Since the tracing feature is optional, it would cause empty configmaps to be created where it is not enabled.

kristinnardal2 replied:
Done,

This works fine for now where we only have the tracing variables in the config map but it'll be more difficult if the condition for creating it becomes more complex.

marcosbc reviewed (changes requested):
Thanks for the contribution! Please have a look at my comments whenever you have a chance.

marcosbc reviewed (approved):
LGTM! Thanks for the contribution!

marcosbc commented:
@kristinnardal2 I'm not able to merge at the moment, because the DCO check failed. Could you check it?

kristinnardal2 commented:
@marcosbc Sorry for that, I've fixed it with 

```bash
$git rebase --exec 'git commit --amend --no-edit -n -s -S' -i HEAD^^
$git push --force
```

I've just switched computers and haven't set everything up properly, I forgot to include the `-s` parameter when I made the last two commits.

marcosbc reviewed (approved):
LGTM! Thanks for the contribution!
